Fix for MacOS keyboard support

psstoyanov opened the pull request:
- Add separate methods to handle raw keyboard events per platform
- Add Mac platform
- Add Linux platform

Notes:
Both Linux and Windows see the events as RawKeyEventDataAndroid at the moment.

psstoyanov committed:
Fix MacOS keyboard support

- Add a separate method to handle raw keyboard events per platform
- Add Mac platform
- Add Linux platform

Notes:
Both Linux and Windows see the events as RawKeyEventDataAndroid at the moment.

tomaszpolanski commented on lib/src/presentation_controller.dart:
```diff
@@ -33,31 +47,64 @@ class PresentationController {
   void _handleKey(RawKeyEvent value) {
     if (value is RawKeyUpEvent) {
+      print(value.data);
+      if (value.data is RawKeyEventDataMacOs) {
```
How about a bit different approach? The main thing we need is `keyCode` from `RawKeyUpEvent`.
How about creating a method that given a `RawKeyEvent` returns us an `int` that is the `keyCode`.
In case that the event cannot be matched, then just return `-1`.
Finally, you would have just a one with a statement that holds all the cases:
```
switch (keyCode) {
      case 20:
      case 21:
      case AndroidKeys.leftKey:
      case AndroidKeys.pageUpKey:
      case AndroidKeys.upKey:
      case MacOSKeys.leftKey:
      case MacOSKeys.pageUpKey:
      case MacOSKeys.upKey:
        _sendAction(PageAction.previous);
        break;
      case 19:
      case 22:
      case AndroidKeys.rightKey:
      case AndroidKeys.pageDownKey:
      case AndroidKeys.downKey:
      case MacOSKeys.rightKey:
      case MacOSKeys.pageDownKey:
      case MacOSKeys.downKey:
        _sendAction(PageAction.next);
        break;
      default:
        break;
    }
```

The only drawback would be that some keys from Android and MacOs would overlap, but for now I do not see that is the case

psstoyanov replied:
Great ideas! Made the changes already.

For the purposes of the library, there shouldn't be a problem with keycode overlap for now.

Maybe some USB device or presentation remote controller could cause such overlaps but that shouldn't be a concern.

psstoyanov committed:
Fix MacOS keyboard support v2

- Use method to return keyCode as int with default value
- Use switch statement to return the keyCode from PlatformData
- Remove duplicated code handling the RawKeyCodes

psstoyanov commented:
Just made an update based on the feedback above.

Nothing complex or attempting to do smart things. Just changing to a switch statement to return the keyCode for the platform and returning it as `int` with default value.

Only Fuschia doesn't have keyCode but I don't see someone using the OS anytime soon. 
Checked on Linux and Windows again just in case - they still use `RawKeyEventDataAndroid`.

I made a rebase from upstream on the fork before pushing the change.

tomaszpolanski commented:
Thanks! I will check it try it out today or tomorrow :)
